functions/go/render-helm-chart/third_party/sigs.k8s.io/kustomize/api/builtins/HelmChartInflationGenerator.go

```diff
@@ -172,39 +183,16 @@ func (p *HelmChartInflationGeneratorPlugin) replaceValuesInline() error {
 	return err
 }
 
-// copyValuesFile to avoid branching.  TODO: get rid of this.
-func (p *HelmChartInflationGeneratorPlugin) copyValuesFile() (string, error) {
-	var b []byte
-	var err error
-	b, err = ioutil.ReadFile(p.ValuesFile)
-	if err != nil {
-		path := p.ValuesFile
-		if u, err := url.Parse(path); err == nil && (u.Scheme == "http" || u.Scheme == "https") {
-			var hc *http.Client
-			hc = &http.Client{}
-
-			resp, err := hc.Get(path)
-			if err != nil {
-				return "", err
-			}
-			defer resp.Body.Close()
-			b, err = ioutil.ReadAll(resp.Body)
-			if err != nil {
-				return "", err
-			}
-		}
-	}
-	return p.writeValuesBytes(b)
-}
-
 // Write a absolute path file in the tmp file system.
 func (p *HelmChartInflationGeneratorPlugin) writeValuesBytes(
-	b []byte) (string, error) {
+	b []byte, path string) (string, error) {
 	if err := p.establishTmpDir(); err != nil {
 		return "", fmt.Errorf("cannot create tmp dir to write helm values")
 	}
-	path := filepath.Join(p.tmpDir, p.Name+"-kustomize-values.yaml")
-	return path, ioutil.WriteFile(path, b, 0644)
+	// use a hash of the provided path to generate a unique, valid filename
+	hash := md5.Sum([]byte(path))
```

Comment:
Is the path an absolute path?

Reply:
Yes, it is an absolute path in the `/tmp` directory.

Reply:
It seems this path is originally passed in by the users. What if they happen to provide a relative path? A safer approach is to resolve to absolute path before compute the hash.

Reply:
Ah, my mistake, I thought you meant the path we are returning here. Fixed, we are now passing in the absolute path into `writeValuesBytes` which is used to generate the hash.